src/VisualStudio/IntegrationTests/AbstractEditorTests.cs

```diff
@@ -0,0 +1,231 @@
+        {
+            return new KeyPress(virtualKey, shiftState);
+        }
+
+        protected void DisableSuggestionMode()
+        {
+            _visualStudioWorkspace.UseSuggestionMode = false;
+        }
+
+        protected void EnableSuggestionMode()
+        {
+            _visualStudioWorkspace.UseSuggestionMode = true;
+        }
+
+        protected void InvokeCompletionList()
+        {
+            ExecuteCommand(WellKnownCommandNames.ListMembers);
+            WaitForAsyncOperations(FeatureAttribute.CompletionSet);
+        }
+
+        protected void ExecuteCommand(string commandName)
+        {
+            _visualStudio.Instance.ExecuteCommand(commandName);
+        }
+
+        protected void VerifyCurrentLineText(string expectedText, bool trimWhitespace = true)
+        {
+            var caretStartIndex = expectedText.IndexOf("$$");
+
+            if (caretStartIndex >= 0)
+            {
+                var caretEndIndex = caretStartIndex + "$$".Length;
+
+                var expectedTextBeforeCaret = caretStartIndex < expectedText.Length
+                    ? expectedText.Substring(0, caretStartIndex)
+                    : expectedText;
+
+                var expectedTextAfterCaret = caretEndIndex < expectedText.Length
+                    ? expectedText.Substring(caretEndIndex)
+                    : string.Empty;
+
+                var lineText = _editor.GetCurrentLineText();
+
+                if (trimWhitespace)
+                {
+                    if (caretStartIndex == 0)
+                    {
+                        lineText = lineText.TrimEnd();
+                    }
+                    else if (caretEndIndex == expectedText.Length)
+                    {
+                        lineText = lineText.TrimStart();
+                    }
+                    else
+                    {
+                        lineText = lineText.Trim();
+                    }
+                }
+
+                var lineTextBeforeCaret = caretStartIndex < lineText.Length
+                    ? lineText.Substring(0, caretStartIndex)
+                    : lineText;
+
+                var lineTextAfterCaret = caretStartIndex < lineText.Length
+                    ? lineText.Substring(caretStartIndex)
+                    : string.Empty;
+
+                Assert.Equal(expectedTextBeforeCaret, lineTextBeforeCaret);
+                Assert.Equal(expectedTextAfterCaret, lineTextAfterCaret);
+                Assert.Equal(expectedTextBeforeCaret.Length + expectedTextAfterCaret.Length, lineText.Length);
+            }
+            else
+            {
+                var lineText = _editor.GetCurrentLineText();
+
+                if (trimWhitespace)
+                {
+                    lineText = lineText.Trim();
+                }
+
+                Assert.Equal(expectedText, lineText);
+            }
+        }
+
+        protected void VerifyTextContains(string expectedText)
```

Comment:
One thing I would like is if functions that take markup actually named their parameters to indicate that.  I wouldn't have ever guessed how this works in the current form. Thanks!

On Jun 23, 2016 4:48 AM, Dustin Campbell <notifications@github.com> wrote:

In src/VisualStudio/IntegrationTests/AbstractEditorTests.cs<https://github.com/dotnet/roslyn/pull/11735#discussion_r68220597>:

> +                Assert.Equal(expectedTextBeforeCaret.Length + expectedTextAfterCaret.Length, lineText.Length);
> +            }
> +            else
> +            {
> +                var lineText = _editor.GetCurrentLineText();
> +
> +                if (trimWhitespace)
> +                {
> +                    lineText = lineText.Trim();
> +                }
> +
> +                Assert.Equal(expectedText, lineText);
> +            }
> +        }
> +
> +        protected void VerifyTextContains(string expectedText)


Well, asserting the caret position is optional, so the method name doesn't seem quite right.

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://github.com/dotnet/roslyn/pull/11735/files/6e2373b9e81d4e6d585c0b9ddd1fd58bbad18aed#r68220597>, or mute the thread<https://github.com/notifications/unsubscribe/AEWmY412XhaUgi7_lfXjNHW0BX_RGTtFks5qOnKjgaJpZM4ItFlT>.